### ISSUE TITLE
Support array of string inputs to the colbert-embedder

### ISSUE THREAD

**Issue by jobergum** [vespa-engine/vespa#30071]
Our recently added [colbert-embedder](https://docs.vespa.ai/en/embedding.html#colbert-embedder) has become very popular, but currently only supports a single string field as input like this:

```
schema doc {
  doc {
    field paragraph type string {..}
  }
  field colbert type tensor<int8>{dt, x[4]} {
   indexing:  input paragraph | embed colbert | attribute
  }
}
```

There has been a considerable demand for support of an array field as input, similar to our `bert-embedder` and `hugging-face-embedder` implementations. A working complete example of how a multi-paragraph version of Colbert would look like, including compression (here using the Vespa-col-minium, which uses 32 floats per vector), compressed to 4 bytes per token ( A pleasant surprise was that `unpack_bits` handles any number of mapped dimensions. Kudos!)  

```
schema doc {

    document doc {
        field paragraphs type array<string> {}
    }
   # Another mapped dimension p represents the paragraph
   field colbert type tensor<int8>(p{}, dt{}, x[4]) {
      indexing: input paragraphs | embed colbert | attribute
   }
    fieldset default {
        fields: paragraphs 
    }

    rank-profile colbert inherits default {
        inputs {
            query(qt) tensor<float>(qt{}, x[32])
         }
        function max_sim() {
            expression {
                sum(
                    reduce(
                        sum(
                            query(qt) * unpack_bits(attribute(colbert)) , x
                        ),
                        max, dt
                    ),
                    qt
                )
            }
        }
        first-phase {
            expression: reduce(max_sim(), max, p)   # a reduce over the paragraphs, envision this to be max
        }
	match-features: max_sim
    }
}
```

## What is needed 
The current colbert embedder implementation expects that the target tensor is mixed with one mapped dimension and one indexed dimension. Similar to our single vector embedders (e.g `bert-embedder`) that expect a single indexed dimension. 

The minimal change is to signal somewhere in configuration (or hardcode it to `p{}` whose mapped dimension is the outer dimension representing the array index.  

Related,  we will soon ship the splade embedder that maps a single text to a mapped tensor; for it to support arrays, we need the same functionality there where the target tensor is then a two-dimensional mapped tensor `tensor<float>(p{}, t{})`

**Comment by jobergum** [comment]
Mock data [tensor playground example](https://docs.vespa.ai/playground/#N4KABGBEBmkFxgNrgmUrWQPYAd5QFNIAaFDSPBdDTAO30gEcBXAgJwE8AKRgFwEoSZGpCIJIvArQDOWNgB5oAGywBDXgD4evYAF9iYAB6IAHAF1+cYAAY4iAIzEATMQDMxACzEArMQBsxADsxOYG9naOLu5evgHBoWBOEc5unj7+QSFmBq7JUWmxmQkeeakxGfFmupDCELrC+sLU5LgMRKQ0UJRotXQM6rxsAJYARsySXAAmAkKdEKIMkjJyiirqWjh6BtNbRqYWVGAAOrTAOHDW27wXBoYXunCOJ2c3YNOvd-YPLs-nl2-Xf53JwPdy-V7vIFwVwPLzg-6Q25wDwPXzwq4fODeB4BdEAzF+B7BPGIoxwQIPEzEE4k66OMnWB5PU5-DH0z7faksiF0pEg6Fcl4I3lkmHIwWs-Hs5GoiU8x5I7FwXHc4UKsmE8lytXSilwKk07n00l3RkK8HGwFIr5wH5GjFQ-lg+34qFiuEuk0yrFyy2YpUql5+qGa4meq1kvUmQ0naq9eoYRoYZqYVridq9Cj4FPkejiAC2qkMAH1pEN87M5gtxNJmPmuCdUGwCJNmABjAgN2gYWv1ljsbh8fhgABUYAGwzGE2mw9u-GIhcMV3nQ5qnQTqCTqBzUDThEr5G6O-meagzdbHeLi+LeA6VbEZ5b7c717L+YXReIOEE8YaKDMIC6EAA)

**Comment by bratseth** [comment]
Done in https://github.com/vespa-engine/vespa/pull/30139

**Comment by bratseth** [comment]
Available in Vespa 8.299.14
